### ISSUE TITLE
InstanceInfoReplicator causes the next Heartbeat to Register the instance

### ISSUE THREAD

**Issue by brenuart** [Netflix/eureka#539]
The following piece of code is executed when the `InstanceInfoReplicator` is triggered:

     if (instanceInfo.isDirty()) {
       register();
       instanceInfo.setIsDirty(false);
     }

`instanceInfo.setIsDirty(false)` has the additional side-effect of setting the `lastDirtyTimestamp` to 'now', so the current value is different from the one transmitted to the Eureka server during the `register()` method.

The next time the `HeartbeatThreat` wakes up, it sends a heartbeat with a dirty timestamp different from what the server got with the last `register()`. Since the dirty timestamp doesn't match what it has in its records, the server replies with a 404 to force the client to re-register with up-to-date information. 

In summary, here is the sequence of operations that occurs every time the `InstanceInfo` is dirtied:

- InstanceInfoReplicator calls `register()`
- HeartbeatThread sends heartbeat and receives 404 from server
- HeartbeatThread sends `register()`

This extra communication could be avoided if `InstanceInfo.setDirty(false)` didn't update the `lastDirtyTimestamp` to current time.

**Comment by qiangdavidliu** [comment]
@brenuart this is already fixed.

**Comment by brenuart** [comment]
Fine. Was working with version 1.1.147 
Sorry about this "false" issue :(
Could you tell me in which version this fix has been made available?

**Comment by qiangdavidliu** [comment]
Sorry should have been more clear. The fix was released in release 1.1.156
